bindings/pyroot/ROOT.py

```diff
@@ -238,6 +240,100 @@ def _TTree__iter__( self ):
 
 _root.CreateScopeProxy( "TTree" ).__iter__    = _TTree__iter__
 
+# Array interface
+def _proxy__array_interface__(self):
+    getter_array_interface = "_get__array_interface__"
+    if hasattr(self, getter_array_interface):
+        return self._get__array_interface__()
+    else:
+        raise Exception("Class {} does not have method {}.".format(
+            type(self), getter_array_interface))
+
+for pyclass in [
+        "std::vector<{dtype}>",
+        "Experimental::VecOps::TVec<{dtype}>"
+        ]:
+    dtypes = ["float", "double", "int", "unsigned int", "long", "unsigned long"]
+    for dtype in dtypes:
+        class_scope = _root.CreateScopeProxy(pyclass.format(dtype=dtype))
+        class_scope._proxy__array_interface__ = _proxy__array_interface__
+        class_scope.__array_interface__ = property(class_scope._proxy__array_interface__)
+
+# TTree.AsMatrix functionality
+def _TTreeAsMatrix(self, columns=None, dtype="double"):
+    # Import numpy lazily
+    try:
+        import numpy as np
+    except:
+        raise ImportError("Failed to import numpy during call of TTree.AsMatrix.")
+
+    # Check that tree has entries
+    if self.GetEntries() == 0:
+        raise Exception("Tree {} has no entries.".format(self.GetName()))
+
+    # Get all columns of the tree if no columns are specified
+    if columns is None:
+        columns = [branch.GetName() for branch in self.GetListOfBranches()]
+
+    # Check validity of branches
```

Comment:
@etejedor I've improved the error-handling so that it checks

1. that the column exists.
2. that the branch has only one leaf with the name of the branch.
3. the data-type of the leaf is supported.

Reply:
I put tests for each case in the unit-test `ttree_asmatrix.py`. However, there is the case that you want to take all columns with `column=None`, but you have a branch that is not supported. Then, it throws an exception instead of skipping the branch. In this case, the user would have to specify the desired branches manually.

Reply:
Thanks @stwunsch ! Regarding the case where `columns=None` and at least one of the branch types is not supported, I wonder if it would be better to skip the unsupported ones and give a warning. That would help when you have many columns and just one or a few are not supported, since you would not have to specify them in a long list. Is this likely? What do the others think @dpiparo @bluehood

Reply:
ah, decisions decisions :sweat_smile: 
I don't have a strong opinion, all options are fine.

If we want this to be as ergonomic as possible, I agree that throwing and forcing the user to manually specify a long list of branches is not nice.

The nicest I can think of is something like this:
```python
# all columns, warn that some get skipped specifying their name and type explicitly
t.AsMatrix()
# just a subset of the columns
t.AsMatrix(("a", "b"))
t.AsMatrix(columns=("a", "b"))
# all columns except a subset
t.AsMatrix(exclude=("obj1", "useless2"))
```

Reply:
All those options makes sense.  You may also have an option that says 'ignore/skip 'errors' branches' [If there is 'several' branches, for the users to arrive the comprehensive list might be tedious when she is just in 'exploring' this file mode]

Reply:
I would definitely refrain from putting any print statements into this function. Imho, nothing worse than methods that pollute stdout without a proper logging system.

So I could add an `exclude` argument and try to make the error-handling fool proof. I'll give it a try. Thanks for the comment!